1259/add compact view in opportunities

abhithory opened the pull request:
Fixed: #1259 #2149 

Here is result:

[Screencast from 22-10-23 06:33:23 AM CDT.webm](https://github.com/twentyhq/twenty/assets/76877003/08fb467e-13b8-41e2-b9d6-dc15e2edb2fc)

charlesBochet commented:
@abhithory Thank you! Looking at the product behavior:
- the Toggle seems too big in the Options Menu. I think we have never reworked properly this component could you add a DropdownSize = 'small' |. 'medium' prop that would take that into account: https://www.figma.com/file/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=6581%3A65983&mode=dev
- in compact mode, the company name (+ eye + checkbox) does not see vertically aligned in the section (could you double check the figma here?)

charlesBochet commented on front/src/modules/companies/components/CompanyBoardCard.tsx:
```diff
@@ -148,6 +193,14 @@ export const CompanyBoardCard = () => {
           />
+          {showCompactViewOptionInCards && (
+            <StyledCompactIconContainer className="compact-icon-container">
+              <StyledIconEye
```
I think we can use showIcon && <StyledIconEye ... /> instead of passing showIcon as a prop of the styled component

charlesBochet commented on front/src/modules/companies/components/CompanyBoardCard.tsx:
```diff
@@ -135,11 +176,15 @@ export const CompanyBoardCard = () => {
     <StyledBoardCardWrapper>
       <StyledBoardCard
         selected={currentCardSelected}
-        onClick={() => setCurrentCardSelected(!currentCardSelected)}
```
I'm surprise we remove this line, not needed anymore?

abhithory replied:
if we don't remove it then the Card get selected when we click on the eye icon in compact mode for expend mode.

is this okay to get selected?? or should i do somethign else?

charlesBochet commented on front/src/modules/companies/components/CompanyBoardCard.tsx:
```diff
@@ -109,6 +142,14 @@ export const CompanyBoardCard = () => {
+  const [isCardInCompactView, setIsCardInCompactView] = useRecoilState(
+    isCardInCompactViewState(boardCardId ?? ''),
+  );
+  const [showCompactViewOptionInCards] = useRecoilState(
```
I would prefer isCompactViewEnabled

charlesBochet commented on front/src/modules/ui/layout/board/components/BoardOptionsDropdownContent.tsx:
```diff
@@ -194,6 +199,21 @@ export const BoardOptionsDropdownContent = ({
           </DropdownMenuItemsContainer>
+          <DropdownMenuSeparator />
+          <DropdownMenuItemsContainer>
+            <MenuItemNavigate
```
We should use a MenuItemToggle instead, and not change MenuItemNavigate behavior

charlesBochet reviewed (commented):
Thanks a lot! Left comments :)

abhithory commented:
ok @Bonapara , i will try

abhithory commented:
@Bonapara @charlesBochet , what should I use for animation? framer motion or normal css?

charlesBochet commented:
@abhithory we prefer to use framer motion by default. We already have @/ui/utilitiies/animation folder that should content components that you can re-use

abhithory commented:
Hi @charlesBochet , Please check now. I have changed the things that you mentioned. also let me know if there is still something to improve.

@Bonapara , I have added the animation also. Please check if it is okay.

[Screencast from 24-10-23 05:47:53 AM CDT.webm](https://github.com/twentyhq/twenty/assets/76877003/e6cbc445-ed39-4420-97a5-2198ce9db63c)

charlesBochet reviewed (approved):
LGTM, thank you!
